src/main/java/org/spongepowered/common/data/SpongeDataManager.java

```diff
@@ -70,15 +68,9 @@
 @Singleton
 public final class SpongeDataManager implements DataManager {
 
-    private static final TypeToken<CatalogType> catalogTypeToken = TypeToken.of(CatalogType.class);
-    private static final TypeToken<DataSerializable> dataSerializableTypeToken = TypeToken.of(DataSerializable.class);
-
     static {
-        TypeSerializers.getDefaultSerializers().registerPredicate(
-            // We have a separate type serializer for CatalogTypes, so we explicitly discount them here.
-            // See https://github.com/SpongePowered/SpongeCommon/issues/1348
-            x -> dataSerializableTypeToken.isSupertypeOf(x) && !catalogTypeToken.isSupertypeOf(x), new DataSerializableTypeSerializer()
-        );
+        // TODO: centralize registration
```

Comment:
Is this something that can be done in sponge, or is it something that needs Configurate 4.x?

Reply:
Most can be done with sponge, but moving from adding to the global collection at various points around Common to adding to a Sponge-specific collection (Following a [Kyori text-like pattern](https://github.com/KyoriPowered/text/blob/master/serializer-gson/src/main/java/net/kyori/text/serializer/gson/GsonComponentSerializer.java#L58)) in one specific spot would be an api-breaking change.

What will change in 4.0 is that modifying existing TypeSerializerCollections will be forbidden when it is currently just discouraged.

Registration could be moved to be in one spot fairly easily, but there isn't much benefit to that when serializers are still being added to the global collection.